lib/posix-process/arch/x86_64/signal.S

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/asm.h>
+
+/* rdi: signum
+ * rsi: siginfo or NULL
+ * rdx: ucontext or NULL
+ * rcx: handler
+ * r8:  app sp
+ */
+ENTRY(pprocess_signal_arch_jmp_handler)
+	/* save uk regs, the application's signal handler
+	 * may corrupt anything. TODO Check calling convention.
+	 */
+	pushq	%rdi
+	pushq	%rsi
+	pushq	%rdx
+	pushq	%rcx
+	pushq	%rax
+	pushq	%r8
+	pushq	%r9
+	pushq	%r10
+	pushq	%r11
+	pushq	%rbx
+	pushq	%rbp
+	pushq	%r12
+	pushq	%r13
+	pushq	%r14
+	pushq	%r15
+	movq	%rsp, %rax	/* temporarily save rsp into rax */
+
+	/* Restore the user stack, user GS, and call the signal handler */
```

Comment:
Yeah, this sequence will probably need some reworking once #1346 gets in :sweat: